### PR TITLE
fix: modifier ordering

### PR DESCRIPTION
**Pull request by atomiks**
I used the second algorithm in the link (the one with most upvotes) instead and adapted it - it seems to work.

I have no idea how it works, so I think we should add some unit tests to make sure it works appropriately?

### CODE REVIEW

**Comment by FezVrasta** [comment]
What’s wrong with the current one?

**Comment by atomiks** [comment]
It wasn't working properly to put it simply 😆  I don't know the specifics as I don't know how it works

**Comment by atomiks** [comment]
The only reason #877 is failing is because `preventOverflow` is being placed before `arrow` again, which happens when I export the `hide` modifier, so it just isn't ordering properly. This one does order properly.

I should probably make sure the tests I added will fail for the previous algo

**Comment by atomiks** [comment]
Sorry for that complete mess 🤦‍♂ 

It should work now lol

File: package.json [hunk 1]
```diff
@@ -88,6 +88,9 @@
       "<rootDir>/src/**/*.test.js",
       "<rootDir>/tests/**/*.test.js"
     ],
+    "globals": {
```
Comment (FezVrasta) [inline package.json]: What’s this for?

Comment (atomiks) [reply package.json]: So Jest doesn't error for files with `__DEV__`

**Comment by atomiks** [comment]
You already are validating dependencies in `requires` it seems in a commit 3 days ago, so you can remove the validation here and the test

**Pull request merged**